libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -598,6 +648,44 @@ bool AP_DDS_Client::create()
             }
         }
     }
+
+    // ROS-2 Service : else case for service requests
+
+    for (size_t i = 0; i < ARRAY_SIZE(services);i++){
+        
+        //status requests
+        constexpr uint8_t nRequests = 1;
+        uint16_t requests[nRequests];
+        constexpr uint16_t requestTimeoutMs = (uint8_t) nRequests * maxTimeMsPerRequestMs;
+        uint8_t status[nRequests];
+
+        if(strlen(services[i].rep_profile_label) > 0){
+            const uxrObjectId rep_id = {
+                .id = services[i].rep_id,
+                .type = UXR_REPLIER_ID
+            };
+            const char* replier_ref = services[i].rep_profile_label;    
+            const auto  replier_req_id = uxr_buffer_create_replier_ref(&session, reliable_out, rep_id, participant_id, replier_ref, UXR_REPLACE);
+
+            requests[0] = replier_req_id;
+
+            if (!uxr_run_session_until_all_status(&session, requestTimeoutMs, requests, status, nRequests)) {
+                GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Service/Replier session request failure for index '%d'",(int)i);
```

Comment:
Why do you cast `i` to `int` when you can just use the size_t unsigned print formatter? 
`printf("%zu\n", x);`